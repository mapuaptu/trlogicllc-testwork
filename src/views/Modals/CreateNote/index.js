import Button from '@/components/Button/index.vue';

export default {
  name: 'ModalCreateNote',
  components: {
    Button,
  },
  data() {
    return {
      message: '',
      noteName: '',
    };
  },
  methods: {
    onCancel() {
      return this.$emit('close');
    },
    onSave() {
      if (!this.noteName) {
        return (this.message = 'Note name requred');
      }

      if (this.$store.state.notes.find((item) => item.name === this.noteName)) {
        return (this.message = 'Note with this name already exist');
      }

      this.$store.commit('CREATE_NOTE', { name: this.noteName });

      return this.$emit('close');
    },
  },
  mounted() {
    return this.$refs.input.focus();
  },
};
