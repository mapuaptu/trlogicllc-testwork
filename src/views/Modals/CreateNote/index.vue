<template>
  <div
    :class="$style.modal"
    class="modal"
  >
    <div class="modal-header">
      {{ title }}
    </div>

    <div class="modal-content">
      <input
        v-model="noteName"
        type="text"
        placeholder="Enter note name"
        class="input"
        @input="message = ''"
      >
      <div
        v-if="message"
        class="message"
      >
        {{ message }}
      </div>
    </div>

    <div class="modal-footer">
      <Button @click.native="onCancel">
        Cancel
      </Button>

      <Button @click.native="onSave">
        Save
      </Button>
    </div>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
