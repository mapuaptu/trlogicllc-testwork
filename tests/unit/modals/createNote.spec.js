import CreateNote from '@/views/Modals/CreateNote/index.vue';
import { mount } from '@vue/test-utils';

describe('Modal createNote component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(CreateNote);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
