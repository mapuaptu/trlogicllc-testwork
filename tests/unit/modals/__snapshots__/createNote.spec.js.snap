// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal createNote component 1. Rendered success 1`] = `
<div class="modal">
  <div class="modal-header">
    Create Note
  </div>
  <div class="modal-content"><input type="text" placeholder="Enter note name" class="input">
    <!---->
  </div>
  <div class="modal-footer">
    <div><span>
      Cancel
    </span>
      <!---->
    </div>
    <div><span>
      Save
    </span>
      <!---->
    </div>
  </div>
</div>
`;
